ipol_demo/modules/demorunner/demorunner.py

```diff
@@ -28,847 +28,713 @@
 
-import cherrypy
 import Tools.build as build
 import Tools.run_demo_base as run_demo_base
 import virtualenv.run as virtualenv
+from fastapi import (
+    APIRouter,
+    Body,
+    Depends,
+    FastAPI,
+    File,
+    HTTPException,
+    Request,
+    UploadFile,
+    status,
+)
+from fastapi.exceptions import RequestValidationError
+from fastapi.responses import JSONResponse, StreamingResponse
+from pydantic import BaseSettings
 from Tools.run_demo_base import IPOLTimeoutError
 
 
-class IPOLMissingBuildItem(Exception):
+class Settings(BaseSettings):
+    default_timeout = 60
+    logs_dir: str = "logs/"
+    authorized_patterns: str = "authorized_patterns.conf"
+
+    compilation_lock_filename: str = "ipol_construct.lock"
+
+    main_bin_dir: str = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), "binaries"
+    )
+    main_log_dir: str = "logs"
+    share_demoExtras_dir: str = (
+        os.path.expanduser("~") + "/ipolDevel/shared_folder/demoExtras"
+    )
+    extra_path: str = (
+        os.path.expanduser("~")
+        + "/ipolDevel/ipol_demo/modules/demorunner/venv/bin:/bin:/usr/bin:/usr/local/bin"
+    )
+    log_file: str = os.path.join(main_log_dir, "error_logs.log")
+
+    config_common_dir: str = (
+        os.path.expanduser("~") + "/ipolDevel/ipol_demo/modules/config_common"
+    )
+    MATLAB_path: str = "/usr/local/matlab/R2015b"
+
+
+settings = Settings()
+app = FastAPI(
+    docs_url="/docs",
+    openapi_url="/openapi.json",
+    redoc_url=None,
+)
+lock_run = Lock()
+
+
+def validate_ip(request: Request) -> bool:
+    # Check if the request is coming from an allowed IP address
+    patterns = []
+    ip = request.headers["x-real-ip"]
+    # try:
+    for pattern in read_authorized_patterns():
+        patterns.append(
+            re.compile(pattern.replace(".", "\\.").replace("*", "[0-9a-zA-Z]+"))
+        )
+    for pattern in patterns:
+        if pattern.match(ip) is not None:
+            return True
+    raise HTTPException(status_code=403, detail="IP not allowed")
+
+
+private_route = APIRouter(dependencies=[Depends(validate_ip)])
+
+
+class IPOLCompileError(Exception):
     """
-    IPOLMissingBuildItem
+    IPOLCompileError
     """
 
 
-class IPOLConstructFileNotFound(Exception):
+def mkdir_p(path):
     """
-    IPOLConstructFileNotFound
+    Implement the UNIX shell command "mkdir -p"
+    with given path as parameter.
     """
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
 
 
-class IPOLUnauthorizedAccess(Exception):
+def init_logging():
     """
-    IPOLUnauthorizedAccess
+    Initialize the error logs of the module.
     """
+    logger = logging.getLogger("core_log")
+    # handle all messages for the moment
+    logger.setLevel(logging.DEBUG)
+    handler = logging.FileHandler(os.path.join(settings.logs_dir, "error.log"))
```

Comment:
```suggestion
    logger = logging.getLogger("demorunner")
    # handle all messages for the moment
    logger.setLevel(logging.DEBUG)
    handler = logging.StreamHandler()
```
so that it prints to stderr (same as `core`, it's better than individual log files)
and remove the `logs_dir` setting